packages/service-library/src/servicelib/logging_utils.py

```diff
@@ -392,3 +392,8 @@ def guess_message_log_level(message: str) -> LogLevelInt:
     ):
         return logging.WARNING
     return logging.INFO
+
+
+def set_parent_module_log_level(current_module: str, desired_log_level: int) -> None:
+    parent_module = ".".join(current_module.split(".")[:-1])
+    logging.getLogger(parent_module).setLevel(desired_log_level)
```

Comment:
NOTE that if parent_module does not exists, it does create the logger

Reply:
Add a small test to see if this actually works. you can use  https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture

Reply:
Very funny I can't write a test that works. But it actually works in the application (I've tested it). I'm unsure why. I can't seem to define two loggers in the same module with a parent relationship.

Reply:
If you have any idea why this tea does not work I'd be glad to hear for you


```python
def test_set_parent_module_log_level(caplog: pytest.LogCaptureFixture):
    caplog.set_level(logging.DEBUG)
    caplog.clear()

    parent = logging.getLogger("parent")
    child = logging.getLogger("parent.child")
    child.propagate = True
    parent.setLevel(logging.WARNING)
    child.setLevel(logging.WARNING)

    child.info("first info")
    assert "first info" not in caplog.text

    set_parent_module_log_level("parent.child", logging.INFO)
    child.info("second info")
    assert "second info" in caplog.text
```

Reply:
SEE https://github.com/pcrespov/sandbox-python/blob/python-stuff/test_logging.py#L18-L69

Reply:
Thanks, figured out what was wrong now.